### PR TITLE
ra-no-code - Introduce Resource Configuration

### PR DESCRIPTION
**Pull request by djhi**
- [x] Introduce Configuration Screen
- [x] Implement basic resource configuration
- [x] Implement basic field configuration
- [x] Allow to select on which views a field is displayed

ra-no-code-configuration

### CODE REVIEW

File: packages/ra-no-code/src/ResourceConfiguration/ResourceConfiguration.tsx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                                            // Here will go an icon selector
+                                            (
+                                                resourceConfiguration.label ||
+                                                resourceConfiguration.name
+                                            ).substr(0, 1)
+                                        }
+                                    </Avatar>
+                                }
+                                action={
+                                    // Will display a menu to delete the resource maybe ?
```
Comment (fzaninotto) [inline packages/ra-no-code/src/ResourceConfiguration/ResourceConfiguration.tsx:85]: Add a FIXME or we'll never revisit that comment

File: packages/ra-no-code/src/ui/useImportResourceFromCsv.tsx [hunk 2]
```diff
@@ -24,23 +25,22 @@ export const useImportResourceFromCsv = (
...
+                    meta
+                );
                 await Promise.all(
-                    data.map(record => {
-                        if (record.id) {
-                            return dataProvider
-                                .create(resource, {
-                                    data: record,
-                                })
-                                .catch(error => console.error(error));
```
Comment (fzaninotto) [inline packages/ra-no-code/src/ui/useImportResourceFromCsv.tsx:37]: why did you remove error catching?

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: packages/ra-core/package.json [hunk 3]
```diff
@@ -66,7 +66,7 @@
     },
     "dependencies": {
         "classnames": "~2.2.5",
-        "date-fns": "^1.29.0",
+        "date-fns": "^2.21.1",
```
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-core/package.json:69]: Ok but then our inference is broken

File: packages/ra-core/src/inference/inferTypeFromValues.ts [hunk 4]
```diff
@@ -170,10 +171,18 @@ export const inferTypeFromValues = (
     }
     if (valuesAreObject(values)) {
         // we need to go deeper
-        // Arbitrarily, choose the first prop of the first object
-        const propName = Object.keys(values[0]).shift();
-        const leafValues = values.map(v => v[propName]);
-        return inferTypeFromValues(`${name}.${propName}`, leafValues);
+        // Arbitrarily, choose the first object
+        // FIXME bad visual representation
```
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-core/src/inference/inferTypeFromValues.ts:175]: I'm not sure, I copied it from yours :)

File: packages/ra-core/src/inference/inferTypeFromValues.ts [hunk 5]
```diff
@@ -170,10 +171,18 @@ export const inferTypeFromValues = (
     }
     if (valuesAreObject(values)) {
         // we need to go deeper
-        // Arbitrarily, choose the first prop of the first object
-        const propName = Object.keys(values[0]).shift();
-        const leafValues = values.map(v => v[propName]);
-        return inferTypeFromValues(`${name}.${propName}`, leafValues);
```
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-core/src/inference/inferTypeFromValues.ts:176]: Don't you want a better inference for objects ?

File: packages/ra-no-code/src/ui/useImportResourceFromCsv.tsx (hunk 2 above)
Comment (djhi) [reply packages/ra-no-code/src/ui/useImportResourceFromCsv.tsx:37]: Because I only used it for debug. A proper error catching should be added

File: packages/ra-no-code/src/ResourceConfiguration/FieldConfigurationFormSection.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    const labelArgs = getFieldLabelTranslationArgs({
+        source: field.props.source,
+        resource,
+        label: field.props.label,
+    });
+
+    return (
+        <CardContent>
+            <TextInput
+                source={`fields[${index}].props.source`}
```
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-no-code/src/ResourceConfiguration/FieldConfigurationFormSection.tsx:22]: I don't get this comment

File: packages/ra-no-code/src/ui/useImportResourceFromCsv.tsx (hunk 2 above)
Comment (djhi) [reply packages/ra-no-code/src/ui/useImportResourceFromCsv.tsx:37]: For now I'd rather have the whole thing explode than the error being hidden like before

File: packages/ra-core/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-core/package.json]: Ignore my previous comment, it works fine

File: packages/ra-no-code/src/ResourceConfiguration/FieldConfigurationFormSection.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-no-code/src/ResourceConfiguration/FieldConfigurationFormSection.tsx:24]: It does not because fields is the array of fields in the resource configuration object

[the remark this answers is not in the document]
Comment (djhi) [reply packages/ra-no-code/src/ResourceConfiguration/FieldConfigurationFormSection.tsx:24]: I'll make it clearer though

**Review by fzaninotto — APPROVED** [review]
